lobster/related.lobster

```diff
@@ -0,0 +1,88 @@
+
+import std
+import dictionary
+import json
+
+let TOPN = 5
+
+struct Post:
+    _id: string
+    title: string
+    tags: [string]
+
+struct RelatedPost:
+    _id: string
+    tags: [string]
+    related: [Post]
+
+def main():
+    let json_content = read_file("../posts.json")
+
+    assert(json_content != nil)
+
+    let posts = JSON.parse<[Post]>(json_content)
+
+    let start = seconds_elapsed()
+
+    let tag_map = dictionary<string, [int]> ( 100 )
+
+    for (posts) post,i:
+        for (post.tags) tag:
+            let indexes = tag_map.get(tag)
+
+            if not indexes:
+                tag_map.set(tag, [i])
+            else:
+                tag_map.set(tag, push(indexes, i))
+
+    var all_related_posts = []::RelatedPost
+    let posts_count = posts.length
+
+    vector_capacity(all_related_posts, posts_count)
+
+    for(posts) post,i:
+
+        let tagged_post_count = map(posts_count) : 0
+
+        for (post.tags) tag:
+            for (assert(tag_map.get(tag))) index:
```

Comment:
Unlike other higher level languages that have a highly tuned dictionary type built-in, Lobster does not, and this implemented in Lobster dictionary is not particularly fast.. but may be amortized if this loop iterates a lot

Reply:
Dictionaries aren't needed but I made them mandatory for a good balance of a real-world scenario, but I will try an impl where tags are represented as int and use a [[int]] instead; to see how much difference it makes. 

Didn't do much better in [go](https://github.com/jinyus/related_post_gen/blob/aff200fe70220d5ed9a2cdf2665abdc347c1f75c/go/main.go#L49) though.